flow/project.py

```diff
@@ -1117,68 +1040,30 @@ def _incomplete(s):
             print(json.dumps(statuses, indent=4), file=file)
             return
 
+        jobs_count = len(statuses)
```

Comment:
Could you try to break up the gathering of all the various template parameters into logical sections and use comments to guide the reader? This line for instance, seems to come a bit out of nowhere.